Source/Core/Core/AchievementManager.cpp

```diff
@@ -1086,6 +1081,54 @@ AchievementManager::PingRichPresence(const RichPresence& rich_presence)
   return r_type;
 }
 
+void AchievementManager::DisplayWelcomeMessage()
+{
+  std::lock_guard lg{m_lock};
+  PointSpread spread = TallyScore();
+  if (hardcore_mode_enabled)
+  {
+    OSD::AddMessage(
+        fmt::format("You have {}/{} achievements worth {}/{} points", spread.hard_unlocks,
+                    spread.total_count, spread.hard_points, spread.total_points),
+        OSD::Duration::VERY_LONG, OSD::Color::YELLOW,
+        (Config::Get(Config::RA_BADGES_ENABLED)) ? GetBadgeTexture(&m_game_badge) : nullptr);
+    OSD::AddMessage("Hardcore mode is ON", OSD::Duration::VERY_LONG, OSD::Color::YELLOW);
+  }
+  else
+  {
+    OSD::AddMessage(fmt::format("You have {}/{} achievements worth {}/{} points",
+                                spread.hard_unlocks + spread.soft_unlocks, spread.total_count,
+                                spread.hard_points + spread.soft_points, spread.total_points),
+                    OSD::Duration::VERY_LONG, OSD::Color::CYAN,
+                    (Config::Get(Config::RA_BADGES_ENABLED)) ? GetBadgeTexture(&m_game_badge) :
+                                                               nullptr);
+    OSD::AddMessage("Hardcore mode is OFF", OSD::Duration::VERY_LONG, OSD::Color::CYAN);
+  }
+}
+
+std::shared_ptr<AbstractTexture>
+AchievementManager::GetBadgeTexture(AchievementManager::BadgeStatus* badge_status)
+{
+  if (badge_status->texture != nullptr)
+    return badge_status->texture;
+  float width = BADGE_HW;
+  float height = BADGE_HW;
+  VideoCommon::CustomTextureData::ArraySlice::Level level;
+  if (!Common::LoadPNG(badge_status->badge, &level.data, &level.width, &level.height))
+  {
+    ERROR_LOG_FMT(ACHIEVEMENTS, "Failed to load texture for badge {}.", badge_status->name);
+    return nullptr;
+  }
+  level.format = AbstractTextureFormat::RGBA8;
```

Comment:
Apologies for the wrong direction, these two lines aren't really necessary after all (they don't hurt but might as well remove them)